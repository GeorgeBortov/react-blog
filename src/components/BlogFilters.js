import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export class BlogFilters extends React.Component {
    state = {
        
    };
    render() {
        return (
            <div className="page-header">
                <div className="content-container">
                    <form>
                        <div className="input-group">
                            <div className="input-group__item">
                                <input
                                    type="text"
                                    className="text-input"
                                    placeholder="Search posts by title"
                                    // value={this.props.filters.text}
                                    // onChange={this.onTextChange}
                                />
                            </div>
                            <div className="input-group__item">
                                <select
                                    className="select"
                                    // value={this.props.filters.sortBy}
                                    // onChange={this.onSortChange}
                                >
                                    <option value="date">Sort by date</option>
                                    <option value="cur_user">Select only my posts</option>
                                </select>
                            </div>
                        </div>
                    </form>
                    <Link className="button" to="/create">Add Post</Link>
                </div>
            </div>
        )
    }
}



export default connect(undefined, undefined)(BlogFilters);