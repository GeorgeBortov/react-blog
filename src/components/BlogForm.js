import React from 'react';
import { connect } from 'react-redux';
import JoditEditor from "jodit-react";
import moment from 'moment';

export class BlogForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            title: props.post ? props.post.title : '',
            body: props.post ? props.post.body : '',
            createdAt: moment(),
            authorName: props.userName ? props.userName : (props.post.authorName ? props.post.authorName : '' ),
            error: ''
        }
    }
    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState(() => ({ title }));
    };
    onAuthorChange = (e) => {
        const authorName = e.target.value;
        this.setState(() => ({ authorName }));
    };
    onBodyChange = (value) => {
        const body = value;
        this.setState(() => ({ body }));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.title || !this.state.body) {
            // 'Please provide description and amount'
            this.setState(() => ({ error: 'Please provide title and post body' }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                title: this.state.title,
                body: this.state.body,
                createdAt: this.state.createdAt.valueOf(),
                authorName: this.state.authorName
            });
        }
    };
    config = {
		readonly: false // all options from https://xdsoft.net/jodit/doc/
	}
    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input
                    type="text"
                    placeholder="Post title"
                    autoFocus
                    className="text-input form--title"
                    value={this.state.title}
                    onChange={this.onTitleChange}
                />
                <JoditEditor
                    editorRef={this.setRef}
                    value={this.state.body}
                    config={this.config}
                    onChange={this.onBodyChange}
                />
                {
                    !this.props.userName &&
                    <input
                        type="text"
                        placeholder="Author"
                        className="text-input form--author"
                        value={this.state.authorName}
                        onChange={this.onAuthorChange}
                    />
                }
                
                <div>
                    <button className="button">Save Post</button>
                </div>
            </form>
        )
    }
}

const mapStateToProps = (state) => ({
    userName: state.auth.name || ''
});

export default connect(mapStateToProps, undefined)(BlogForm);