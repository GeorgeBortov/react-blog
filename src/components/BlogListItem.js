import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import moment from 'moment';
import trimText from '../selectors/trim-text';
export const BlogListItem = ({ id, title, trimText, authorName, createdAt, uid, authorID, }) => (
    <div className="list-item">
        <div>
            <h3 className="list-item__title">{title}</h3>
            <div className="list-item__text">
                {trimText}
            </div>
        </div>
        <div className="list-item__data">
            <span className="list-item__author">Posted by {authorName !== '' ? authorName : 'Anonymous'}</span>
            <span className="list-item__time">{moment(createdAt).format('MMMM Do YYYY')} at {moment(createdAt).format('h:mm')}</span>
        </div>
        <div className="list-item__buttons">
            <Link className="button" to={`/read/${id}`}>Read</Link>
            {   
                uid === authorID &&
                    <Link className="button" to={`/edit/${id}`}>Edit</Link>
                
            }
            
        </div>
    </div>
    
);

const mapStateToProps = (state, ownProps) => ({
    uid: state.auth.uid,
    trimText: trimText(ownProps.body, 500)
});

const ConnectedBlogListItem = connect(mapStateToProps)(BlogListItem);

export default ConnectedBlogListItem;