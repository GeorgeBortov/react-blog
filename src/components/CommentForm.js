import React from 'react';
import { connect } from 'react-redux';
import JoditEditor from "jodit-react";
import moment from 'moment';

export class BlogForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            rate: props.post ? props.post.body : '',
            body: props.post ? props.post.body : '',
            createdAt: moment(),
            authorName: props.userName,
            error: ''
        }
        this.baseState = this.state
    }
    handleRateChange = (e) => {
        const rate = e.target.value;
        this.setState(() => ({ rate })); 
    };
    onAuthorChange = (e) => {
        const authorName = e.target.value;
        this.setState(() => ({ authorName }));
    };
    onBodyChange = (value) => {
        const body = value;
        this.setState(() => ({ body }));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.authorName || !this.state.body || !this.state.rate) {
            // 'Please provide description and amount'
            this.setState(() => ({ error: 'Please fill in all fields and rate the post' }));
        } else {
            e.target.reset();
            this.props.onSubmit({
                rate: this.state.rate,
                body: this.state.body,
                createdAt: this.state.createdAt.valueOf(),
                authorName: this.state.authorName
            });
            this.setState(this.baseState);    
        }
    };
    config = {
		readonly: false // all options from https://xdsoft.net/jodit/doc/
	}
    render() {
        return (
            <div className="comment-form">
                
                <h4>Write a commpent</h4>
                
                <form className="form" onSubmit={this.onSubmit}>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <span className="rating">
                        <input type="radio" id="star5" name="rating" value="5" onBlur={this.handleRateChange}  /><label htmlFor="star5" title="Rocks!">5 stars</label>
                        <input type="radio" id="star4" name="rating" value="4" onBlur={this.handleRateChange}  /><label htmlFor="star4" title="Pretty good">4 stars</label>
                        <input type="radio" id="star3" name="rating" value="3" onBlur={this.handleRateChange}  /><label htmlFor="star3" title="Meh">3 stars</label>
                        <input type="radio" id="star2" name="rating" value="2" onBlur={this.handleRateChange}  /><label htmlFor="star2" title="Kinda bad">2 stars</label>
                        <input type="radio" id="star1" name="rating" value="1" onBlur={this.handleRateChange}  /><label htmlFor="star1" title="Sucks big time">1 star</label>
                        <span className="txt">Please rate:</span>
                    </span>
                    {
                        !this.props.userName &&
                        <input
                            type="text"
                            placeholder="Your name"
                            className="text-input form--author"
                            value={this.state.authorName}
                            onChange={this.onAuthorChange}
                        />
                    }
                    <JoditEditor
                        editorRef={this.setRef}
                        value={this.state.body}
                        config={this.config}
                        onChange={this.onBodyChange}
                    /> 
                    <div>
                        <button className="button">Publish</button>
                    </div>
                </form>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    userName: state.auth.name || state.auth.email
});

export default connect(mapStateToProps, undefined)(BlogForm);