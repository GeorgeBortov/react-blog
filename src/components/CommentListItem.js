import React from 'react';
import moment from 'moment';
import { Markup } from 'interweave';
export const CommentListItem = ({ authorName, body, createdAt, rate }) => (
    <div className="list-item">
        <div>
            <div className="star-ratings-wrap">
                <div className="star-ratings">
                    <span style={{width: rate*100/5 + '%'}} className="star-ratings-sprite-rating"></span>
                </div>
                <span className="score">Rating: {rate}/5</span>
            </div>
            <div className="list-item__text">
                <Markup content={body} />
            </div>
        </div>
        <div className="list-item__data">
            <span className="list-item__author">Posted by {authorName !== '' ? authorName : 'Anonymous'}</span>
            <span className="list-item__time">{moment(createdAt).format('MMMM Do YYYY')} at {moment(createdAt).format('h:mm')}</span>
            
        </div>
    </div>
    
);


export default CommentListItem;