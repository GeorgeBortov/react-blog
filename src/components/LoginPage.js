import React from 'react';
import { connect } from 'react-redux';
import { 
    startRegisterUser,
    startEmailLogin,
    startGoogleLogin,
    sendResetLink,
    authWithNewAccount
} from '../actions/auth';
import ConfirmAuthModal from './ConfirmAuthModal';
import ModalInfo from './ModalInfo';
import { openModal, closeModal } from '../actions/modal';

export class LoginPage extends React.Component {
    constructor(props) {
        super(props);
        this._toggleDiv = this._toggleDiv.bind(this)
        this.state = {
            email: props.email ? props.email : '',
            pass: props.pass ? props.pass : '',
            emailReset: props.emailReset ? props.emailReset : ''
        };
    };

    _toggleDiv() {
        $(this.refs['toggle-div']).slideToggle();
    }

    onInputChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }
    addNewAccount = () => {
        this.props.closeModal();
        this.props.authWithNewAccount(this.state.email, this.state.pass);
    }
    loginWithEmail = (e) => {
        e.preventDefault();
        this.props.startEmailLogin(this.state.email, this.state.pass)
            .catch(error => {
                // console.log(error.code);
                if (!this.state.email || !this.state.pass) {
                    this.setState(() => ({
                        emailError: !this.state.email && 'Please provide an email',
                        passwordError: !this.state.pass && 'Please provide a password',
                        emailResetError: undefined,
                        userOrPasswordError: undefined
                    }));
                } else {
                    if (error.code === 'auth/invalid-email') {
                        this.setState(() => ({
                            emailError: 'Please provide a valid email address',
                            emailResetError: undefined,
                            passwordError: undefined,
                            userOrPasswordError: undefined
                        }));
                    }
                    if (error.code === 'auth/user-not-found') {
                        this.setState(() => ({
                            userOrPasswordError: 'The email you\'ve entered doesn\'t match any account',
                            emailError: undefined,
                            emailResetError: undefined,
                            passwordError: undefined
                        }));
                    }
                    if (error.code === 'auth/wrong-password') {
                        this.setState(() => ({
                            userOrPasswordError: 'Email or password is incorrect',
                            emailError: undefined,
                            emailResetError: undefined,
                            passwordError: undefined
                        }));
                    }
                }
            });
    };
    regWithEmail = (e) => {
        e.preventDefault();
        this.props.startRegisterUser(this.state.email, this.state.pass)
            .catch(error => {
                // console.log(error.code);
                if (!this.state.email || !this.state.pass) {
                    this.setState(() => ({
                        emailError: !this.state.email && 'Please provide an email',
                        passwordError: !this.state.pass && 'Please provide a password',
                        emailResetError: undefined,
                        userOrPasswordError: undefined
                    }));
                } else {
                    if (error.code === 'auth/invalid-email') {
                        this.setState(() => ({
                            emailError: 'Please provide a valid email address',
                            emailResetError: undefined,
                            passwordError: undefined,
                            userOrPasswordError: undefined
                        }));
                    }
                    if (error.code === 'auth/weak-password') {
                        this.setState(() => ({
                            passwordError: 'Password should be at least 6 characters',
                            emailError: undefined,
                            emailResetError: undefined,
                            userOrPasswordError: undefined
                        }));
                    }
                    if (error.code === 'auth/email-already-in-use') {
                        this.props.openModal();
                    }
                }
            });
    };
    resetPass = (e) => {
        e.preventDefault();
        this.props.sendResetLink(this.state.emailReset)
            .then(() => {
                this.setState(() => ({
                    userOrPasswordError: undefined,
                    emailResetError: undefined,
                    emailError: undefined,
                    passwordError: undefined,
                    email: '',
                    pass: '',
                    emailReset: '',
                    modalIsOpenInfo: true
                }));
            })
            .catch(error => {
                // console.log(error.code);
                if (!this.state.emailReset) {
                    this.setState(() => ({
                        emailError: undefined,
                        emailResetError: !this.state.emailReset && 'Please provide an email',
                        passwordError: undefined,
                        userOrPasswordError: undefined
                    }));
                } else {
                    if (error.code === 'auth/invalid-email') {
                        this.setState(() => ({
                            emailResetError: 'Please provide a valid email address',
                            emailError: undefined,
                            passwordError: undefined,
                            userOrPasswordError: undefined
                        }));
                    }
                    if (error.code === 'auth/user-not-found') {
                        this.setState(() => ({
                            userOrPasswordError: 'The email you\'ve entered doesn\'t match any account',
                            emailResetError: undefined,
                            emailError: undefined,
                            passwordError: undefined
                        }));
                    }
                }
            });
    };

    closeInfoModal = (e) => {
        this.setState(() => ({
            modalIsOpenInfo: false
        }));
    };

    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box box-layout--login">
                    <h1 className="box-layout__title">Log In</h1>
                    <button className="button button--google-login" onClick={this.props.startGoogleLogin}>
                        <span className="button--soc-icon">
                            <img src="/images/google-ico.png" />
                        </span>
                        <span>
                            Login with Google
                        </span>
                    </button>
                    <div className="box-layout__divider">or</div>
                    <form>
                        <div className={!this.state.userOrPasswordError ? "form-group total" : "form-group error total"}>
                            <span className="error-message total">{this.state.userOrPasswordError}</span>
                        </div>
                        <div className={!this.state.emailError ? "form-group" : "form-group error"}>
                            <input
                                type="text"
                                className={this.state.email.length > 0 ? "form-control edited" : 'form-control'}
                                name="email"
                                autoComplete="off"
                                value={this.state.email}
                                onChange={this.onInputChange}
                            />
                            <label>Email Address</label>
                            <span className="error-message">{this.state.emailError}</span>
                        </div>
                        <div className={!this.state.passwordError ? "form-group" : "form-group error"}>
                            <input
                                type="password"
                                name="pass"
                                className={this.state.pass.length > 0 ? "form-control edited" : 'form-control'}
                                value={this.state.pass}
                                autoComplete="off"
                                onChange={this.onInputChange}
                            />
                            <label>Password</label>
                            <span className="error-message">{this.state.passwordError}</span>
                        </div>
                        <div>
                            <button className="button button--login-email" onClick={this.loginWithEmail}>
                                <span className="button--soc-icon">
                                    <svg aria-hidden="true" data-prefix="fas" data-icon="at" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z"></path></svg>
                                </span>
                                <span>
                                    Login with Email
                                </span>
                            </button>
                        </div>
                        <div>
                            <button className="button button--reg-email" onClick={this.regWithEmail}>
                                <span className="button--soc-icon">
                                    <svg aria-hidden="true" data-prefix="fas" data-icon="at" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z"></path></svg>
                                </span>
                                <span>
                                    Registrate with Email
                                </span>
                            </button>
                        </div>
                        <div className="my-collapsible">
                            <a className="button button--link" onClick={this._toggleDiv}>Forgot password?</a>
                            <div className="toggle-div" ref="toggle-div">
                                Please provide the email address you would like a password reset link sent to.
                                <div className={!this.state.emailResetError ? "form-group" : "form-group error"}>
                                    <input
                                        type="text"
                                        className={this.state.emailReset.length > 0 ? "form-control edited" : 'form-control'}
                                        name="emailReset"
                                        autoComplete="off"
                                        value={this.state.emailReset}
                                        onChange={this.onInputChange}
                                    />
                                    <label>Email Address</label>
                                    <span className="error-message">{this.state.emailResetError}</span>
                                </div>
                                <div>
                                    <button className="button button--resetPass" onClick={this.resetPass}>
                                        Reset
                                    </button>
                                </div>
                            </div>
                        </div>
                    </form>
                </div>
                <ConfirmAuthModal
                    isOpen={this.props.modalStatusConfirm}
                    onRequestClose={this.props.closeModal}
                    addNewAccount={this.addNewAccount}
                />
                <ModalInfo
                    isOpen={this.state.modalIsOpenInfo}
                    onRequestClose={this.closeInfoModal}
                />
            </div>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    startRegisterUser: (email, password) => dispatch(startRegisterUser(email, password)),
    startEmailLogin: (email, password) => dispatch(startEmailLogin(email, password)),
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    sendResetLink: (email) => dispatch(sendResetLink(email)),
    authWithNewAccount: (email, password) => dispatch(authWithNewAccount(email, password)),
    openModal: () => dispatch(openModal()),
    closeModal: () => dispatch(closeModal())
});

const mapStateToProps = (state, props) => ({
    modalStatusConfirm: state.modal.modalIsOpen
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);