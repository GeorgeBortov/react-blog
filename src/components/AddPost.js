import React from 'react';
import { connect } from 'react-redux';
import BlogForm from './BlogForm';
import { startAddPost } from '../actions/posts';

export class AddPost extends React.Component {
    onSubmit = (post) => {
        this.props.startAddPost(post);
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add post</h1>
                    </div>
                </div>
                <div className="content-container">
                    <BlogForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddPost: (post) => dispatch(startAddPost(post))
});

export default connect(undefined, mapDispatchToProps)(AddPost);