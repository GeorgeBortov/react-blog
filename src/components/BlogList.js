import React from 'react';
import { connect } from 'react-redux';
import BlogListItem from './BlogListItem';


export const BlogList = (props) => {
    return (
        <div className="content-container">
            <div className="list-body">
                {   
                    props.posts.length === 0 ? (
                        <div className="list-item list-item--message">
                            <span>No posts</span>
                        </div>
                    ) : (
                        props.posts.map((post) => {
                            return <BlogListItem key={post.id} {...post} />
                        })
                    )
                }
            </div>
        </div>
    )
};

const mapStateToProps = (state) => ({
    posts: state.posts
});

const ConnectedBlogList = connect(mapStateToProps)(BlogList);

export default ConnectedBlogList;