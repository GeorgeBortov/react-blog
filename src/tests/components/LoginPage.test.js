import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
    const startGoogleLogin = jest.fn();
    const wrapper = shallow(<LoginPage startGoogleLogin={startGoogleLogin}/>);
    wrapper.find('.button--google-login').simulate('click', {
        preventDefault: () => { }
    });
    expect(startGoogleLogin).toHaveBeenCalled();
});

test('should call startEmailLogin on button click', () => {
    const startEmailLogin = jest.fn(() => Promise.reject(new Error()));
   
    const wrapper = shallow(<LoginPage startEmailLogin={startEmailLogin} />);
    wrapper.find('.button--login-email').simulate('click', {
        preventDefault: () => { }
    });

    expect(startEmailLogin).toHaveBeenCalled();

});

test('should call regWithEmail on button click', () => {
    const startRegisterUser = jest.fn(() => Promise.reject(new Error()));
   
    const wrapper = shallow(<LoginPage startRegisterUser={startRegisterUser} />);
    wrapper.find('.button--reg-email').simulate('click', {
        preventDefault: () => { }
    });

    expect(startRegisterUser).toHaveBeenCalled();

});

test('should call resetPass on button click', () => {
    const sendResetLink = jest.fn(() => Promise.reject(new Error()));
   
    const wrapper = shallow(<LoginPage sendResetLink={sendResetLink} />);
    wrapper.find('.button--resetPass').simulate('click', {
        preventDefault: () => { }
    });

    expect(sendResetLink).toHaveBeenCalled();

});
