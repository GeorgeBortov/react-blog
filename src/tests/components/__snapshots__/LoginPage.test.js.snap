// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box box-layout--login"
  >
    <h1
      className="box-layout__title"
    >
      Log In
    </h1>
    <button
      className="button button--google-login"
    >
      <span
        className="button--soc-icon"
      >
        <img
          src="/images/google-ico.png"
        />
      </span>
      <span>
        Login with Google
      </span>
    </button>
    <div
      className="box-layout__divider"
    >
      or
    </div>
    <form>
      <div
        className="form-group total"
      >
        <span
          className="error-message total"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          autoComplete="off"
          className="form-control"
          name="email"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label>
          Email Address
        </label>
        <span
          className="error-message"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          autoComplete="off"
          className="form-control"
          name="pass"
          onChange={[Function]}
          type="password"
          value=""
        />
        <label>
          Password
        </label>
        <span
          className="error-message"
        />
      </div>
      <div>
        <button
          className="button button--login-email"
          onClick={[Function]}
        >
          <span
            className="button--soc-icon"
          >
            <svg
              aria-hidden="true"
              data-icon="at"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Login with Email
          </span>
        </button>
      </div>
      <div>
        <button
          className="button button--reg-email"
          onClick={[Function]}
        >
          <span
            className="button--soc-icon"
          >
            <svg
              aria-hidden="true"
              data-icon="at"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C118.941 8 8 118.919 8 256c0 137.059 110.919 248 248 248 48.154 0 95.342-14.14 135.408-40.223 12.005-7.815 14.625-24.288 5.552-35.372l-10.177-12.433c-7.671-9.371-21.179-11.667-31.373-5.129C325.92 429.757 291.314 440 256 440c-101.458 0-184-82.542-184-184S154.542 72 256 72c100.139 0 184 57.619 184 160 0 38.786-21.093 79.742-58.17 83.693-17.349-.454-16.91-12.857-13.476-30.024l23.433-121.11C394.653 149.75 383.308 136 368.225 136h-44.981a13.518 13.518 0 0 0-13.432 11.993l-.01.092c-14.697-17.901-40.448-21.775-59.971-21.775-74.58 0-137.831 62.234-137.831 151.46 0 65.303 36.785 105.87 96 105.87 26.984 0 57.369-15.637 74.991-38.333 9.522 34.104 40.613 34.103 70.71 34.103C462.609 379.41 504 307.798 504 232 504 95.653 394.023 8 256 8zm-21.68 304.43c-22.249 0-36.07-15.623-36.07-40.771 0-44.993 30.779-72.729 58.63-72.729 22.292 0 35.601 15.241 35.601 40.77 0 45.061-33.875 72.73-58.161 72.73z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Registrate with Email
          </span>
        </button>
      </div>
      <div
        className="my-collapsible"
      >
        <a
          className="button button--link"
          onClick={[Function]}
        >
          Forgot password?
        </a>
        <div
          className="toggle-div"
        >
          Please provide the email address you would like a password reset link sent to.
          <div
            className="form-group"
          >
            <input
              autoComplete="off"
              className="form-control"
              name="emailReset"
              onChange={[Function]}
              type="text"
              value=""
            />
            <label>
              Email Address
            </label>
            <span
              className="error-message"
            />
          </div>
          <div>
            <button
              className="button button--resetPass"
              onClick={[Function]}
            >
              Reset
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
  <ConfirmModal
    addNewAccount={[Function]}
  />
  <ModalInfo
    onRequestClose={[Function]}
  />
</div>
`;
